core/src/main/scala/org/bitcoins/core/crypto/TxSigComponent.scala

```diff
@@ -53,6 +53,20 @@ sealed abstract class BaseTxSigComponent extends TxSigComponent {
   override def sigVersion = SigVersionBase
 }
 
+sealed abstract class P2SHTxSigComponent extends BaseTxSigComponent {
```

Comment:
how does is this handled relative to `WitnessTxSigComponentP2SH` ? It seems that they are essentially the same wrapper around a `redeemScript`. Any way we can share code here?

Reply:
There is some code re-use but not sure if it is possible to share code here.  `WitnessTxSigComponentP2SH` cannot extend `P2SHTxSigComponent` because it needs to extend WitnessTxSigComponent`